server/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -72,6 +79,10 @@ public void setDocument(int docId) {
 
     @Override
     public ScriptDocValues<?> get(Object key) {
+        // deprecate _type
+        if ("_type".equals(key)) {
+            DEPRECATION_LOGGER.deprecatedAndMaybeLog(TYPES_DEPRECATION_MESSAGE, TYPES_DEPRECATION_MESSAGE);
```

Comment:
The first argument to deprecatedAndMaybeLog should be an arbitrary key uniquely identifying this deprecation point. Normally it would be something like `type-field-doc-lookup`.

Reply:
Fixed.